scripts/mkcloud

```diff
@@ -675,6 +675,11 @@ function restartcloud()
 # bring up VMs that will take cloud controller/compute/storage roles
 function setupnodes()
 {
+    local nodenumbercontroller=1
+    if [[ $clusterconfig == *"services"* ]]; then
+        nodenumbercontroller=`echo ${clusterconfig} | sed -e "s/^.*services[^:]*=\([[:digit:]]\+\).*/\1/"`
```

Comment:
@jdsn I guess you don't need persuading, but anyway it's worth noting that this is another good example to add to [the list](https://github.com/search?utf8=%E2%9C%93&q=shellsucks+&type=Issues&ref=searchresults) of why #shellsucks ;-)  `clusterconfig` collapses a complex data structure (essentially a hash or associative array, mapping cluster names to cluster sizes) into a single scalar value.  Then in places like this we have to resort to disgustingly foul hacks like the above in order to extract the cluster size.  If this file was in Python or Ruby, this line would instead have been:

    nodenumbercontroller = clusterconfig['services']

which is ever so slightly nicer ;-)

Of course bash 4 supports associative arrays, so if `clusterconfig` was made into one we could do

    nodenumber=${clusterconfig[services]}

but then `clusterconfig` could not be passed in between `mkcloud` and `qa_crowbarsetup.sh`.

Just in case there's any doubt, I am not at all criticising @abel-navarro here ;-)  As the code currently stands, this hack is unavoidable :-(